src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -218,13 +213,96 @@ public Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document document,
             return result.ToImmutableAndFree();
         }
 
-        public async Task<CodeFixCollection> GetFixesAsync(Document document, TextSpan range, string diagnosticId, CancellationToken cancellationToken)
+        public async Task<CodeFixCollection> GetFixesInProcAsync(Document document, TextSpan range, string diagnosticId, CancellationToken cancellationToken)
```

Comment:
I think we should use something like GetDocumentFixAllForIdInSpan or something. 

in proc is not the main point here (even though right now it is, we might move inproc to out of proc later)

Reply:
Also, why should we run this in proc?  It seems like that will generate a lot of garbage in the IDE that then has to be cleaned up (and GC's impact other IDE features).  Why not run these out of proc, but still just compute *only* for the single doc, and not all docs?

Reply:
we might in future, but all diagnostics for open file run in proc. it is not doing something special.

Reply:
Well.. it's special in taht it's now running N things :)    But, fair enough.  Something to consider for the future :)